core/src/main/java/hudson/model/User.java

```diff
@@ -702,10 +703,15 @@ private static File getRootDir() {
      * prevent anyone from logging in as these users. Therefore, we prevent
      * saving a User with one of these ids.
      *
-     * @return true if the username or fullname is valid
+     * @param id ID to be checked
+     * @return {@code true} if the username or fullname is valid.
+     *      For {@code null} or blank IDs returns {@code false}.
      * @since 1.600
      */
-    public static boolean isIdOrFullnameAllowed(String id) {
+    public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
+        if (id == null || StringUtils.isBlank(id)) {
```

Comment:
I'd better keep it as is and let Java optimize it.
But I can remove the check if you feel strongly